sim/team-validator.js

```diff
@@ -921,6 +921,24 @@ class Validator {
 		while (template && template.species && !alreadyChecked[template.speciesid]) {
 			alreadyChecked[template.speciesid] = true;
 			if (dex.gen === 2 && template.gen === 1) tradebackEligible = true;
+			// STABmons Move Legality (for tournaments)
+			// @ts-ignore
+			let restrictedMoves = format.restrictedMoves || [];
+			if (ruleTable.has('stabmonsmovelegality') && !restrictedMoves.includes(move.name) && !move.isZ) {
+				let types = template.types;
+				let baseTemplate = dex.getTemplate(template.baseSpecies);
+				if (baseTemplate.otherFormes) {
+					for (let i in baseTemplate.otherFormes) {
```

Comment:
In case you don't understand the TypeScript error, the problem is that `otherFormes` is an array, not an object, and should be iterated with `for..of`, not `for..in`.

Reply:
It was not iterable because it was possibly `null`. Now that you added `if (baseTemplate.otherFormes)` it should be iterable again.

Reply:
I'm not trying to be condescending, I thought you actually didn't understand the TypeScript error.

Reply:
For what it's worth, though, the error was from the lack of a surrounding `if (baseTemplate.otherFormes)` due to the chance of it not existing, not because I used a for...in loop.

Reply:
(The last comment was written before you posted those)